stages/test/test_mkfs_ext4.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/python3
+
+import os.path
+import re
+import subprocess
+from unittest import mock
+
+import pytest
+
+import osbuild.meta
+from osbuild.testutil import has_executable
+from osbuild.testutil.imports import import_module_from_path
+
+
+@pytest.mark.parametrize("test_data,expected_err", [
+    # bad
+    ({}, "'uuid' is a required property"),
+    ({"uuid": 123}, "123 is not of type 'string'"),
+    ({"uuid": "vaild", "label": "12345678901234567"}, " is too long"),
+    ({"uuid": "valid", "verity": "please"}, "'please' is not of type 'boolean'"),
+    # good
+    ({"uuid": "some", "label": "1234567890123456"}, ""),
+    ({"uuid": "some", "label": "label", "verity": True}, ""),
+    # actually "some-uuid" will not be accepted by mkfs, it has to be a valid
+    # uuid but our schema is not strict enough right now
+    ({"uuid": "some-uuid"}, ""),
+])
+def test_schema_validation_mkfs_ext4(test_data, expected_err):
+    name = "org.osbuild.mkfs.ext4"
+    root = os.path.join(os.path.dirname(__file__), "../..")
+    mod_info = osbuild.meta.ModuleInfo.load(root, "Stage", name)
+    schema = osbuild.meta.Schema(mod_info.get_schema(version="2"), name)
+
+    test_input = {
+        "type": "org.osbuild.mkfs.ext4",
+        "devices": {
+            "device": {
+                "path": "some-path",
+            },
+        },
+        "options": {
+        }
+    }
+    test_input["options"].update(test_data)
+    res = schema.validate(test_input)
+
+    if expected_err == "":
+        assert res.valid is True, f"err: {[e.as_dict() for e in res.errors]}"
+    else:
+        assert res.valid is False
+        assert len(res.errors) == 1, [e.as_dict() for e in res.errors]
+        err_msgs = [e.as_dict()["message"] for e in res.errors]
+        assert expected_err in err_msgs[0]
+
+
+@pytest.mark.skipif(not has_executable("mkfs.ext4"), reason="need mkfs.ext4")
```

Comment:
Nitpick alert :) We should probably standardize on the wording for missing executables in these tests. Right now we have 4 different ways to say it. I'm partial to the form I used `no mkfs.ext4 executable` of course, but I'm more a fan of consistency.

Reply:
Thank you! I like this attention to detail and agree. I have no strong opinion here, I will follow your lead here and use `no {name} executable` - at least I will try to remember, please call me out if I forget :)